### PR TITLE
[Docs] Migrate docs framework to Docusaurus

### LINKED ISSUE #165: [Docs] Motivation and goals
Thank you so much for this project!

The project description says Woodpecker is "a highly opinionated fork" of Drone CI, but neither docs nor the repos mention what those opinions are, what the difference to Drone is and why one would choose one over the other.

The original motivation is laid out in now archived https://github.com/laszlocph/drone-oss, but I think it could be very helpful for those new to either with a brief FAQ in the readme and/or docs

### LINKED ISSUE #254: Documentation site out of date
I noticed the documentation site is out of date. 

See: https://woodpecker.laszlo.cloud/server-setup/

Is there plans to update this on this domain or move it to GitHub pages?

### LINKED ISSUE #255: Clear Statement where we differ from drone
Add a statement to the FAQ in docks and Readme since this question is asked frequently ...

### LINKED ISSUE #268: Add list of supported VCS and their limitations
Add a list of supported version control system (github, gitea, ...) to the docs with a list for each of them including the specific limitations (and maybe features) of  them.
For example: Gitlab - not supporting multi-pipeline builds

### PR DESCRIPTION
**Pull request by anbraten**
closes #254 
closes #268 
closes #165
closes #255

> Note: Some parts are marked with TODO as they are based on ongoing changes from other PRs or I simply don't know what to write at that point :wink:. I would suggest to leave them for know and create new PRs updating them later.

# Todo
- [x] replace mkdocs with docosaurus (improved menu structure, ...)
- [x] structure docs into `Usage` and `Server Setup / Administration`
- [x] update favicon
- [x] Create new pipeline-syntax page with all options and links to more detailed docs if available :tada: :tada: :tada:
- [x] add ci to publish to `woodpecker-ci.github.io`
- [x] deploy docs preview to surge for review
- [x] update start-page

# preview

https://probable-stop.surge.sh

# screenshots
Screenshot from 2021-08-29 09-39-47

# required actions by maintainer
- [x] create some bot account for github?
- [x] create some ssh key and save to woodpecker as `GITHUB_PRIVATE_KEY` 

[Docs](https://docusaurus.io/docs/deployment#using-drone) about deploy workflow

### CODE REVIEW

**Comment by anbraten** [comment]
@6543 I updated the docs a "bit" and added some deployment workflow. For that we need some github user with a private ssh key applied to woodpecker to automatically deploy docs. I had the idea that some `woodpecker-bot` account would be best suited for that. 

Could you have a look to the docs `cd docs/; yarn install; yarn start`. 

At the start page we could list some features of woodpecker. What do you think would be suited or should we replace it with something else?

**Comment by 6543** [comment]
Will take me some time

File: docs/docs/10-intro.md [hunk 1]
```diff
@@ -2,16 +2,16 @@
 
 Woodpecker is a simple CI engine with great extensibility.
 
-![woodpecker](/images/woodpecker.png)
+![woodpecker](woodpecker.png)
 
-## .drone.yml
+## .woodpecker.yml
```
Comment (inverse) [inline docs/docs/10-intro.md:7]: Is this really the case? When did this change? see: https://github.com/woodpecker-ci/woodpecker/pull/154

Comment (6543) [reply docs/docs/10-intro.md:7]: I think we support both - in the log term I would say we should use `.woodpecker.yml` as the two projects can devide but use `.drone.yml` as fallback - if nothing is set in settings (settings overwrite it in all cases) ...

Comment (anbraten) [reply docs/docs/10-intro.md:7]: Changing to some woodpecker related name is planned for 0.15.0 #133, but maybe it was a bit to quick to update it here already.

Comment (6543) [reply docs/docs/10-intro.md:7]: no leave it ... but we have to resolve #133 first then :)

File: docs/docs/20-usage/21-multi-pipeline.md [hunk 2]
```diff
@@ -1,11 +1,10 @@
+# Multi pipelines
 
-# Multi-pipeline builds
+> NOTE: This Feature is only available for Github & Gitea repositories. Follow [this](https://github.com/woodpecker-ci/woodpecker/issues/131) issue to support further development.
```
Comment (inverse) [inline docs/docs/20-usage/21-multi-pipeline.md:3]:
```suggestion
> NOTE: This Feature is only available for GitHub & Gitea repositories. Follow [this](https://github.com/woodpecker-ci/woodpecker/issues/131) issue to support further development.
```

Might want to find and replace as there are lots of cases of this in the existing documentation too.

File: docs/docs/20-usage/21-multi-pipeline.md [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
 
-**NOTE: Feature is only available for Github repositories. Follow [this](https://github.com/woodpecker-ci/woodpecker/issues/131) issue to support further development.**
+By default, Woodpecker looks for the pipeline definition in `.woodpecker.yml` in the project root.
 
-By default, Woodpecker looks for the pipeline definition in `.drone.yml` in the project root.
-
-The Multi-Pipeline feature allows the pipeline to be splitted to several files and placed in the `.drone/` folder
+The Multi-Pipeline feature allows the pipeline to be split to several files and placed in the `.woodpecker/` folder. Only `.yml` will we used and files in any subfolders like `.woodpecker/sub-folder/test.yml` will be ignored.
```
Comment (inverse) [inline docs/docs/20-usage/21-multi-pipeline.md:7]: Isn't this configurable from the UI on a per repo basis too?

**Review by inverse — COMMENTED** [review]
Some feedback - looks nice though 👏🏼

File: docs/docs/20-usage/21-multi-pipeline.md (hunk 3 above)
Comment (anbraten) [reply docs/docs/20-usage/21-multi-pipeline.md:7]: That's correct you can change `.woodpecker/` to some custom path in the project settings. 

```suggestion
The Multi-Pipeline feature allows the pipeline to be split into several files and placed in the `.woodpecker/` folder. Only `.yml` files will we used and files in any subfolders like `.woodpecker/sub-folder/test.yml` will be ignored. You can set some custom path like `.my-ci/pipelines/` instead of `.woodpecker/` in the [project settings](/docs/usage/project-settings). 
```

File: docs/docs/20-usage/21-multi-pipeline.md (hunk 2 above)
Comment (anbraten) [reply docs/docs/20-usage/21-multi-pipeline.md]: I changed it for the complete documentation.

**Comment by 6543** [comment]
I'll go with github pages but in the long run a own domain would be best ...

File: docs/docs/91-migrations.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Migrations
+
+Some versions need some changes to the server configuration or the pipeline configuration files.
+
+## 0.15.0
+
+- Default pipeline path changed to `.woodpecker/`
+
+  **Solution:** Rename your `.drone.yml` to `.woodpecker/<some name of your choice>.yml` or enable the fallback flag in [project settings](/docs/usage/project-settings#pipeline-path).
```
Comment (6543) [inline docs/docs/91-migrations.md:9]:
```suggestion
  **Solution:** Rename your `.drone.yml` to `.woodpecker.yml` or enable the fallback flag in [project settings](/docs/usage/project-settings#pipeline-path).
  
  There is still a default fallback mechanism in following order: `.woodpecker/*.yml` -> `.drone.yml` -> `.drone/*.yml`
```

File: remote/mocks/Remote.go [hunk 5]
```diff
@@ -1,11 +1,15 @@
-// Code generated by mockery v1.0.0. DO NOT EDIT.
+// Code generated by mockery v0.0.0-dev. DO NOT EDIT.
```
Comment (6543) [inline remote/mocks/Remote.go:2]: this is unrelated please move it into it's own pull!!

File: docs/src/components/HomepageFeatures.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import clsx from 'clsx';
+import styles from './HomepageFeatures.module.css';
+
+const FeatureList = [
+  {
+    title: 'OpenSource and free',
+    Svg: require('../../static/img/undraw_docusaurus_mountain.svg').default,
```
Comment (6543) [inline docs/src/components/HomepageFeatures.js:8]: wee need own pictures i think ...

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+```
+$ yarn build
+```
+
+This command generates static content into the `build` directory and can be served using any static contents hosting service.
+
+### Deployment
+
+```
+$ GIT_USER=<Your GitHub username> USE_SSH=true yarn deploy
```
Comment (6543) [inline docs/README.md:30]: I would use a seperate repo: https://github.com/woodpecker-ci/pages

Comment (anbraten) [reply docs/README.md:30]: Docusaurus uses its config `organizationName: 'woodpecker-ci', projectName: 'woodpecker-ci.github.io',` to push to a repo called `woodpecker-ci.github.io` in the `woodpecker-ci` org. [See docs](https://docusaurus.io/docs/deployment#docusaurusconfigjs-settings)

File: remote/mocks/Remote.go (hunk 5 above)
Comment (anbraten) [reply remote/mocks/Remote.go]: That was some incorrect merge. I reverted it.

File: docs/README.md (hunk 7 above)
Comment (6543) [reply docs/README.md:30]: INFO to all reviewers: changed repo to https://github.com/woodpecker-ci/woodpecker-ci.github.io ;)

File: docs/LICENSE [hunk 8]
```diff
@@ -1 +0,0 @@
-Files in this folder are licensed under Creative Commons Attribution-ShareAlike 4.0 International Public License. It is a derivative work of the https://github.com/drone/docs git repository.
```
[the remark this answers is not in the document]
Comment (anbraten) [reply docs/LICENSE]: That was by mistake as I removed the old docs folder.  Should / can we change it to match the main projects license?

**Comment by 6543** [comment]
what's the difference to `docs/static/img/logo.svg` and `docs/static/img/favicon.svg` ?

File: docs/sidebars.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  /*
+  tutorialSidebar: [
+    {
+      type: 'category',
+      label: 'Tutorial',
+      items: ['hello'],
+    },
+  ],
+   */
+};
```
Comment (6543) [inline docs/sidebars.js:26]:
```suggestion
module.exports = {
  // let Docusaurus generates a sidebar from the docs folder structure
  tutorialSidebar: [{type: 'autogenerated', dirName: '.'}],
};
```

**Comment by anbraten** [comment]
> what's the difference to `docs/static/img/logo.svg` and `docs/static/img/favicon.svg` ?

I don't think there is one. Just replaced the content of the sample files from Docusaurus.

File: .woodpecker/release-docs.yml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    secrets:
+      - GITHUB_PRIVATE_KEY
+    environment:
+      - USE_SSH=true
+      - GIT_USER=woodpecker-bot
+      - DEPLOYMENT_BRANCH=master
+    commands:
+      - mkdir -p $HOME/.ssh
+      - ssh-keyscan -t rsa github.com >> $HOME/.ssh/known_hosts
+      - echo "$GITHUB_PRIVATE_KEY > $HOME/.ssh/id_rsa"
```
Comment (6543) [inline .woodpecker/release-docs.yml:19]:
```suggestion
      - echo "$BOT_PRIVATE_KEY > $HOME/.ssh/id_rsa"
```

Comment (anbraten) [reply .woodpecker/release-docs.yml]: updated

File: docs/docs/91-migrations.md (hunk 4 above)
Comment (6543) [reply docs/docs/91-migrations.md:9]: @anbraten do you have better wording for this?

File: docs/src/components/HomepageFeatures.js (hunk 6 above)
Comment (anbraten) [reply docs/src/components/HomepageFeatures.js]: I replaced the images with some free icons.

**Review by 6543 — APPROVED** [review]
